### PR TITLE
Allow custom CircleCI Steps for software test execution pipeline

### PR DESCRIPTION
**Pull request by dborowiecki**
TICKET: [TECH-992](https://govstack-global.atlassian.net/browse/TECH-992)

[TECH-992]: https://govstack-global.atlassian.net/browse/TECH-992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/templates/generate-config.py [hunk 1]
```diff
@@ -31,6 +31,27 @@ def is_valid(test_harness_path):
...
+
+def load_circle_config(application_path):
+    """
+    Loads custom CircleCI configuration from 'circle_config.yml' if available.
+    """
+    config_path = os.path.join(application_path, 'circle_config.yml')
+    if os.path.isfile(config_path):
+        with open(config_path, 'r') as file:
+            return yaml.safe_load(file)
+    return None
```
Comment (lruzicki) [inline src/templates/generate-config.py:82]: Is the user informed when the file is present but in an incorrect format? Can we add any validation here to inform the user about issues with additional steps?
we could implement something like this:
 ```python 
 schema = { ... }
 data = yaml.safe_load
 try:
            jsonschema.validate(instance=data, schema=schema)
 catch validation error
...
 ```

**Review by lruzicki — CHANGES_REQUESTED** [review]
Looks good, I have only two questions about the scheme.

Can we ensure thorough documentation or readme for users on how to create and use circle_config.yml? Maybe some example data will be enough or we can skip it as the code is well described

File: src/templates/generate-config.py (hunk 1 above)
Comment (dborowiecki) [reply src/templates/generate-config.py:82]: That's great remark, I didn't add json validation as it'd require installing extra package. However, I've added in code validation for before and after formats. Let me know if that will work for you,

**Review by lruzicki — APPROVED** [review]
LGTM

**Pull request merged**